src/metrics-collector/cf/cf_client_test.go

```diff
@@ -9,217 +10,245 @@ import (
+				It("should error", func() {
+					Expect(err).To(MatchError(MatchRegexp("Error requesting endpoints: .*")))
 				})
 			})
 
-			Context("when the request fails", func() {
+		})
+
+		Context("when retrieving endpoints succeeds", func() {
+			Context("when the auth url is not valid", func() {
 				BeforeEach(func() {
-					fakeCC.Close()
-					fakeCC = nil
+					fakeCC.AppendHandlers(
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", PATH_CF_INFO),
+							ghttp.RespondWith(200, bytes.Replace(infoBody, []byte("test-oauth-endpoint"), []byte("http://www.not-exist.com"), -1)),
+						),
+					)
 				})
 
-				It("returns an error", func() {
+				It("should have the endpoints populated and return error", func() {
+					Expect(cfc.GetEndpoints().AuthEndpoint).To(Equal("http://www.not-exist.com"))
+					Expect(cfc.GetEndpoints().TokenEndpoint).To(Equal("test-token-endpoint"))
+					Expect(cfc.GetEndpoints().DopplerEndpoint).To(Equal("test-doppler-endpoint"))
+
 					Expect(err).To(BeAssignableToTypeOf(&url.Error{}))
 					urlErr := err.(*url.Error)
 					Expect(urlErr.Err).To(BeAssignableToTypeOf(&net.OpError{}))
-				})
-			})
-		})
-
-		/**
-		Context("when retrieving endpoints succeed", func() {
-			BeforeEach(func() {
-				fakeCC.AppendHandlers(
-					ghttp.CombineHandlers(
-						ghttp.VerifyRequest("GET", "/internal/bulk/apps", "batch_size=2&format=fingerprint&token={}"),
-						ghttp.VerifyBasicAuth("the-username", "the-password"),
-						ghttp.RespondWith(200, `{
-								"token": {"id":"the-token-id"},
-								"fingerprints": [
-									{
-										"process_guid": "process-guid-1",
-										"etag": "1234567.890"
-									},
-									{
-										"process_guid": "process-guid-2",
-										"etag": "2345678.901"
-									}
-								]
-							}`),
-					),
-				)
-			})
 
-			It("retrieves the endpoints", func() {
+				})
 
 			})
 
-			Context("when grant with password", func() {
+			Context("when the auth url is valid", func() {
 				BeforeEach(func() {
-					conf = fakes.FakeCfConfig
-					conf.Api = testApiUrl
-					conf.GrantType = GRANT_TYPE_PASSWORD
+					fakeCC.AppendHandlers(
+						ghttp.CombineHandlers(
+							ghttp.VerifyRequest("GET", PATH_CF_INFO),
+							ghttp.RespondWith(200, bytes.Replace(infoBody, []byte("test-oauth-endpoint"), []byte(fakeLoginServer.URL()), -1)),
+						),
+					)
 				})
 
-				It("should not error, and get correct tokens and endpoints", func() {
-					Expect(err).To(BeNil())
-					Expect(cfc.GetEndpoints().AuthEndpoint).To(Equal(testAuthUrl))
-					Expect(cfc.GetEndpoints().DopplerEndpoint).To(Equal(fakes.FAKE_DOPPLER_ENDPOINT))
-					Expect(cfc.GetEndpoints().TokenEndpoint).To(Equal(fakes.FAKE_TOKEN_ENDPOINT))
-					Expect(cfc.GetTokens().AccessToken).To(Equal(fakes.FAKE_ACCESS_TOKEN))
-					Expect(cfc.GetTokens().RefreshToken).To(Equal(fakes.FAKE_REFRESH_TOKEN))
-				})
-			})
+				Context("when login server fails", func() {
```

Comment:
The Login server is not running which is different than a fail (status code != 200).